internal/scms/github/git_operations.go

```diff
@@ -64,11 +72,37 @@ func (gh GitAuthenticationProvider) GetUser(ctx context.Context) (string, error)
 	return "git", nil
 }
 
-// GetClient creates a new GitHub client using the provided SCM provider and secret.
-func GetClient(scmProvider v1alpha1.GenericScmProvider, secret v1.Secret) (*github.Client, error) {
-	itr, err := ghinstallation.New(http.DefaultTransport, scmProvider.GetSpec().GitHub.AppID, scmProvider.GetSpec().GitHub.InstallationID, secret.Data[githubAppPrivateKeySecretKey])
+// GetClientFromInstallationId creates a new GitHub client with the specified installation ID.
+func getClientFromInstallationId(scmProvider v1alpha1.GenericScmProvider, secret v1.Secret, id int64) (*github.Client, *ghinstallation.Transport, error) {
+	itr, err := ghinstallation.New(http.DefaultTransport, scmProvider.GetSpec().GitHub.AppID, id, secret.Data[githubAppPrivateKeySecretKey])
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create GitHub installation transport: %w", err)
+	}
+
+	var client *github.Client
+	if scmProvider.GetSpec().GitHub.Domain == "" {
+		client = github.NewClient(&http.Client{Transport: itr})
+	} else {
+		baseURL := fmt.Sprintf("https://%s/api/v3", scmProvider.GetSpec().GitHub.Domain)
+		itr.BaseURL = baseURL
+		uploadsURL := fmt.Sprintf("https://%s/api/uploads", scmProvider.GetSpec().GitHub.Domain)
+		client, err = github.NewClient(&http.Client{Transport: itr}).WithEnterpriseURLs(baseURL, uploadsURL)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to create GitHub enterprise client: %w", err)
+		}
+	}
+
+	return client, itr, nil
```

Comment:
```suggestion
		return client, itr, nil
	}
	baseURL := fmt.Sprintf("https://%s/api/v3", scmProvider.GetSpec().GitHub.Domain)
	itr.BaseURL = baseURL
	uploadsURL := fmt.Sprintf("https://%s/api/uploads", scmProvider.GetSpec().GitHub.Domain)
	client, err = github.NewClient(&http.Client{Transport: itr}).WithEnterpriseURLs(baseURL, uploadsURL)
	if err != nil {
		return nil, nil, fmt.Errorf("failed to create GitHub enterprise client: %w", err)
	}
	return client, itr, nil
```

Just switches to return early instead of nesting.